stdlib/src/builtin/string_literal.mojo

```diff
@@ -346,3 +346,36 @@ struct StringLiteral(
             An integer value that represents the string, or otherwise raises.
         """
         return _atol(self)
+
+    fn format_simple(self, **kwargs: String) raises -> String:
+        """Build a `String` with a template and variadic keyword arguments.
+
+        Example:
+
+        ```mojo
+        var x = "Is {number} even? {result}".format_simple(
+            number=2, result=True
+        )
+        print(x) #Is 2 even? True
+        ```
+
+        Args:
+            kwargs: Keyword/value pairs passed as variadic keyword arguments.
+
+        Returns:
+            A `String` built by replacing `{keyword}` with `kwargs[keyword]`.
+
+        Values need to implement `Stringable`.
+
+        If any of the replacement values contains curly braces,
+
+        an error is raised: `"A replacement value contains {}"`.
+
+        """
+
+        var res = String(self)
```

Comment:
Since you're creating a string anyways. It would be preferable to implement the logic in `String` itself and then here simply do something like this so we can have this functionality in both types.
```mojo
var res = String(self)
return res.format_simple(kwargs)
```

Reply:
Yes, thanks to both of you for the great suggestion :+1: